### PR TITLE
enhancement(file source): Log IO errors when globbing

### LINKED ISSUE #2420: Log when Vector cannot list / glob directories due to permissions errors
From our chat:

> The globbing in the file source, can you use multiple asterisks in multiple locations. E.g is this valid? include = ["/opt/nomad/data/alloc/*/alloc/logs/monitor.std*.*"]
>
> To answer my own question: yes. The problem I was having was that Vector could not see these files because the "alloc" dir was owned root:root with mode 0711, which meant the vector user couldn't get a directory listing, therefore the globbing failed. I feel like this should have been logged by Vector, but it wasn't.

### PR DESCRIPTION
**Pull request by ktff**
Closes #2420 

### Todo

- [x] Update docs

### CODE REVIEW

File: src/internal_events/file.rs [hunk 1]
```diff
@@ -261,6 +261,29 @@ mod source {
...
+    #[derive(Debug)]
+    pub struct PathGlobbingFailed<'a> {
+        pub path: &'a Path,
+        pub error: &'a Error,
+    }
+
+    impl<'a> InternalEvent for PathGlobbingFailed<'a> {
+        fn emit_logs(&self) {
+            error!(
+                message = "Failed globbing path.",
```
Comment (jszwedko) [inline src/internal_events/file.rs:273]:
```suggestion
                message = "Failed to glob path.",
```

File: src/internal_events/file.rs [hunk 2]
```diff
@@ -261,6 +261,29 @@ mod source {
...
+                path = ?self.path,
+                error = ?self.error,
```
Comment (jszwedko) [inline src/internal_events/file.rs:275]: Should we be using the `Display` trait here and for `path` per https://github.com/timberio/vector/blob/master/CONTRIBUTING.md#logging-style? That is: `path = %self.path, error = %self.error`.

**Review by jszwedko — APPROVED** [review]
Thanks @ktff ! This seems sensible to me.

File: docs/reference/components/sources/internal_metrics.cue [hunk 3]
```diff
@@ -726,5 +734,9 @@ components: sources: internal_metrics: {
 			required:    true
 			default:     "vector"
 		}
+		_path: {
+			description: "The path that produced the error"
```
Comment (pablosichert) [inline docs/reference/components/sources/internal_metrics.cue:738]:
```suggestion
			description: "The path that produced the error."
```

**Review by pablosichert — APPROVED** [review]
LGTM (with minor suggestion)

File: src/internal_events/file.rs (hunk 2 above)
Comment (ktff) [reply src/internal_events/file.rs:275]: Yep, I thought `Path` only had `Debug` but the compiler was smarter and suggested `.display()` method.

Comment (ktff) [reply src/internal_events/file.rs:275]: I'll also update the rest of the events in the file to follow this.

**Review by lukesteensen — APPROVED** [review]

**Pull request merged**